src/Conversion/ONNXToKrnl/Math/Reduction.cpp

```diff
@@ -438,19 +412,16 @@ struct ONNXReduceSumOpLowering : public ConversionPattern {
           !llvm::dyn_cast<ONNXReduceSumOp>(op).noop_with_empty_axes()) {
         IndexExprScope axesloopContex(&rewriter, loc);
         MemRefBoundsIndexCapture axesBounds(axesVal);
-        auto zeroIndex = rewriter.create<arith::ConstantIndexOp>(loc, 0);
-        auto cond =
-            rewriter.create<arith::CmpIOp>(loc, arith::CmpIPredicate::eq,
-                axesBounds.getDim(0).getValue(), zeroIndex);
-        initVal =
-            rewriter.create<arith::SelectOp>(loc, cond, trueVal, falseVal);
+        Value zeroIndex = rewriter.create<arith::ConstantIndexOp>(loc, 0);
```

Comment:
please use this one here and elsewhere (grepping for `ConstantIndexOp`) 

```
  struct MathBuilder {...
  Value constantIndex(int64_t val) const;
  }
```

Reply:
done